presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -551,7 +552,34 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         if (needPostProjectionCoercion) {
             return explicitCoercionFields(subPlan, distinctGroupingColumns, analysis.getAggregates(node));
         }
-        return subPlan;
+
+        // 4. Project and re-write all grouping functions
+        return handleGroupingOperations(subPlan, node, groupIdSymbol);
+    }
+
+    private PlanBuilder handleGroupingOperations(PlanBuilder subPlan, QuerySpecification node, Optional<Symbol> groupIdSymbol)
+    {
+        if (analysis.getGroupingOperations(node).isEmpty()) {
+            return subPlan;
+        }
+
+        TranslationMap translations = subPlan.copyTranslations();
```

Comment:
call it `newTranslations` and don't use it for translations in
```
for (GroupingOperation groupingOperation : analysis.getGroupingOperations(node)) {
```
loop since those translations are new and not yet projected.